lib/flight_job/models/script.rb

```diff
@@ -59,9 +59,18 @@ def self.load_all
       end.reject(&:nil?)
     end
 
-    attr_writer :id, :notes
+    attr_reader :id
+    attr_writer :notes
+
+    validates :id, presence: true, length: { maximum: FlightJob.config.max_id_length },
+              format: { with: /\A[a-zA-Z0-9_-]+\Z/,
+                        message: 'must be alphanumeric but may contain hyphen and underscore' }
+    validates :id, format: { with: /\A[[:alnum:]].*\Z/, message: 'must start with an alphanumeric character' }
```

Comment:
For consistency with the above, "must start with a letter or a number".

Reply:
65f796c08732a8fd277b7bc00162554fca0a0d2f